code/game/objects/structures/tables_racks.dm

```diff
@@ -45,6 +46,13 @@
 	update_adjacent()
 	..()
 
+/obj/structure/table/proc/checkhealth()
```

Comment:
I went over this with @9600bauds .  Running it as a table/glass/proc/checkhealth() caused it to wig out and not recognize the proc as defined.  Running it at the parent level with an &&istype glass table worked and with the sanity from making sure its a glass table will prevent any sort of fuckery with destroying non glass tables.